AuthenticatorShared/Core/Platform/Services/NotificationCenterService.swift

```diff
@@ -0,0 +1,36 @@
+import Combine
+import UIKit
+
+// MARK: - NotificationCenterService
+
+/// A protocol for a `NotificationCenterService` which accesses the app's notification center.
+///
+protocol NotificationCenterService: AnyObject {
+    /// A publisher for when the app enters the background.
+    ///
+    func didEnterBackgroundPublisher() -> AnyPublisher<Void, Never>
+
+    /// A publisher for when the app enters the foreground.
+    ///
+    func willEnterForegroundPublisher() -> AnyPublisher<Void, Never>
+}
+
+// MARK: - DefaultNotificationCenterService
+
+/// A default implementation of the `NotificationCenterService` which accesses the app's notification center.
+///
+class DefaultNotificationCenterService: NotificationCenterService {
+    func didEnterBackgroundPublisher() -> AnyPublisher<Void, Never> {
+        NotificationCenter.default
+            .publisher(for: UIApplication.didEnterBackgroundNotification)
+            .map { _ in }
+            .eraseToAnyPublisher()
+    }
+
+    func willEnterForegroundPublisher() -> AnyPublisher<Void, Never> {
+        NotificationCenter.default
+            .publisher(for: UIApplication.willEnterForegroundNotification)
+            .map { _ in }
+            .eraseToAnyPublisher()
```

Comment:
you could take this a step further and inject the NotificationCenter.default. Then have very basic tests to just make sure the MockNotificationCenter publishes through this publisher.

Reply:
It would be good to backport those tests/implementation to the PM app at some point, if we do.

Reply:
Good suggestion. I added a commit with tests to cover this and I have a similar set of tests in a branch for the PM app that I'll put up after this is approved. 👍